code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -593,6 +593,17 @@
 	if(candidates.len)
 		theghost = pick(candidates)
 		spawn_guardian(user, theghost.key)
+		for(var/mob/living/simple_animal/hostile/guardian/G in living_mob_list)
```

Comment:
https://github.com/HippieStationCode/HippieStation13/blob/7fb5bbbfcc6e1b6880f70c837ea9617df509b4d8/code/modules/mob/living/simple_animal/guardian/guardian.dm#L659
Just add a "return G" so this proc will return the guardian, then you edit the line up here(   spawn_guardian(user, theghost.key)   ) to var/mob/living/simple_animal/hostile/guardian/G = spawn_guardian(user, theghost.key)